### DISCUSSION TITLE
math: rendering not supported in GitHub wiki

### DISCUSSION THREAD

**Discussion by ayala-io** [community/community#17157] [Other Feature Feedback, Questions, & Ideas] [answered]
The built-in GitHub wiki still doesn't support rendering math as is now possible with Markdown files in repositories. It's one reason we still use a repository of Markdown files for our own wiki instead of the GitHub wiki. Any chance we will see this feature ported into GitHub wikis soon?

**Comment by nschloe** [comment]
Good call. If you're so inclined, you could use my browser extension [xhub](https://github.com/nschloe/xhub) for the time being. I think it's even _better_ than GitHub's native math support, but you do have to install the extension once.

**Reply by ayala-io** [reply]
Thank you for the suggestion. I didn't know about xhub. Will try it out. I'd prefer to use the built-in wiki pages in the end if it supported all the features I needed. 

As the developer of xhub, is there a reason why you specifically need the math to be written as:

````markdown
Display math:

```math
e^{i\pi} + 1 = 0
```

and line math $`a^2 + b^2 = c^2`$.
````
instead of using `$ a^2 + b^2 = c^2 $` and `$$ e^{i\pi} + 1 = 0 $$` for inline and block math respectively? This would make the Markdown written specifically to render the math using your extension to be incompatible with the new GitHub support.

**Reply by UlyssesZh** [reply]
> instead of using `$ a^2 + b^2 = c^2 $` and `$$ e^{i\pi} + 1 = 0 $$` for inline and block math respectively? This would make the Markdown written specifically to render the math using your extension to be incompatible with the new GitHub support.

The browser extension was developed before the new GitHub feature.
If we do not use code blocks to write TeX, similar problems to #17143, #16939, and #16993 will rise.

**Comment by imagingbook** [comment]
This was/is really a much needed feature and I was happy to finally see math support this year come after such a long wait. But why is it that this works in GitHub's Markdown files only but is NOT enabled in Wiki pages? I cannot find a note anywhere mentioning this restriction! Hope this will be fixed soon!

**Reply by nschloe** [reply]
I don't think it'll be fixed quickly. You could use [xhub](https://github.com/nschloe/xhub) as a drop-in replacement; it works on Wiki pages, too.

**Reply by imagingbook** [reply]
Yes, I'm afraid so. But xhub requires that all clients must have this extension installed to see rendered math, no?

**Reply by nschloe** [reply]
That's right.

**Comment by davidstaheli** [comment, marked as answer]
This is now supported in wikis: [Wikis now support math and Mermaid diagrams](https://github.blog/changelog/2022-08-09-wikis-now-support-math-and-mermaid-diagrams/). But please also check out @nschloe's excellent [https://github.com/nschloe/xhub](xhub) extension!

**Reply by ayala-io** [reply]
That's great to hear, thank you!

**Reply by nschloe** [reply]
Very simple math constructs like `$\{\}$` still fail to render correctly, unfortunately.
